be/src/exec/vectorized/aggregate/agg_hash_variant.h

```diff
@@ -402,14 +402,29 @@ struct AggHashMapVariant {
         return 0;
     }
 
-    size_t memory_usage() const {
+    size_t reserved_memory_usage(const MemPool* pool) const {
         switch (type) {
 #define M(NAME)      \
     case Type::NAME: \
-        return NAME->hash_map.dump_bound();
+        return NAME->hash_map.dump_bound() + pool->total_reserved_bytes();
+
+            APPLY_FOR_AGG_VARIANT_ALL(M)
+#undef M
+        }
+
+        return 0;
+    }
+
+    size_t allocated_memory_usage(const MemPool* pool) const {
+        switch (type) {
+#define M(NAME)      \
+    case Type::NAME: \
+        return sizeof(decltype(NAME)::element_type::KeyType) * NAME->hash_map.size() + pool->total_allocated_bytes();
```

Comment:
Why `type_size * entries + total_allocated_bytes()` ?

Reply:
There are two kinds of key:
1. A pointer `char*`, which memory is allocated from `mem_pool`.
2. A trivial struct or a int, which memory is allocated from operator system.

Therefore,
- for the first type, `mem_pool-> total_allocated_bytes()` is the real key memory usage, and `type_size` is the pointer memory usage.
- for the second type, `mem_pool-> total_allocated_bytes()` is 0, and `type_size` is the real key memory usage.